Setup/Product.wxs

```diff
@@ -1581,6 +1599,11 @@
       <ComponentRef Id="Microsoft.VisualStudio.Threading.dll" />
       <ComponentRef Id="Microsoft.VisualStudio.Validation.dll" />
       <ComponentRef Id="System.Runtime.InteropServices.RuntimeInformation.dll" />
+      <ComponentRef Id="EasyHook.dll" />
+      <ComponentRef Id="EasyHook32.dll" />
+      <ComponentRef Id="EasyHook64.dll" />
+      <ComponentRef Id="win10default.colors" />
+      <ComponentRef Id="dark.colors" />
```

Comment:
Are these getting deployed to the app's installation folder (i.e. under Program Files)? If so, users won't be able to modify them because the folder is write-protected.
Themes should be deployed to the `LocalAppDataOrganizationDir` folder.

Reply:
My motivation was to have non-editable "technical" theme *win10default.colors* in application directory, which is only supposed to be used in color transformation calculations.

Moving to user data directory is done at runtime by `ThemeDeployment`, which should copy themes other than *win10default.colors* to user data directory.

Reply:
Now that you asked, I doubt it's a good idea to do at runtime, because it's indeed installation-specific detail, so should be implemented in installer.

Reply:
In https://github.com/gitextensions/gitextensions/pull/7562 I am removing possibility to edit theme from GitExtensions, until it's reimplemented as plugin.

Maybe it's ok to have all themes non-editable until then?

Reply:
If we put a theme under `C:\Users\<user>\AppData\Roaming\GitExtensions\GitExtensions` will GE load it?
If so, I think we should update the installer to put it in the right place. Otherwise we will be generating unnecessary work for us in the future.

Reply:
The installer is already configured to install files under `LocalAppDataOrganizationDir` (e.g. the PluginManager`, I think it should be relatively easy to send the themes to that location (or `LocalAppDataOrganizationDir\Themes`).

Reply:
>Now that you asked, I doubt it was a good idea to do at runtime

Except if you have in mind the support of themes in the portable version...

Reply:
The installer has nothing to do with the portable version though. The portable version is created by a different script.
But your note is important to keep in mind - theme files could be in the app's folder (for a portable version) or under the AppData folder (an installed version).

Reply:
This user data path distinction between app / portable is supported in GitExtensions codebase,
https://github.com/gitextensions/gitextensions/blob/ca9124627be4fea07a380a6a01725763ea585fa5/GitCommands/Settings/AppSettings.cs#L80

Reply:
👍 thank you.
It is late, can't think straight now.

So let's update the installer to deploy to the AppData\Roaming\GitExtensions\GitExtensions and we're good to go

Reply:
I prefer to have the default themes in the installation, otherwise may custom changes be overwritten at updates.

Reply:
What I did initially to preserve user changes:
- copy from installation directory to AppData\Roaming
- backup any existing theme files
- backup in such a way that previous backups are not lost

I agree with @RussKie it turned out overly complicated.

I suggest to take following approach:

All theme files are installed to installation directory, they are not supposed to be changed by user.

When user edits the theme and saves the changes, they are saved in writeable AppData directory.

When we present themes list in dropdown, we somehow indicate the difference between built-in themes and custom user themes.

This way
- user's work is safe, customized theme files are never silently overwritten
- no need to spam hard drive with backups
- user is unlikely to accidentally overwrite built-in themes because they lie in Installation directory, while
save / load file dialog will point to AppData by default.

Reply:
> This way

Optimal solution in my opinion.
What we have is maybe good enough though.

Reply:
Thank you all, good points, it makes sense.